drivers/ms5611/MS5611.cpp

```diff
@@ -329,66 +335,78 @@ int MS5611::_collect(uint32_t *raw)
 	ret = _readReg(cmd, &buf[0], 3);
 
 	if (ret < 0) {
-		*raw = 0;
+		raw = 0;
 		return -1;
 	}
 
 	cvt.b[0] = buf[2];
 	cvt.b[1] = buf[1];
 	cvt.b[2] = buf[0];
 	cvt.b[3] = 0;
-	*raw = cvt.w;
+	raw = cvt.w;
 
 	return 0;
 }
 
 void MS5611::_measure(void)
 {
+	if (_measure_phase == 0) {
+		if (_collect(temperature_from_sensor) < 0) {
+			DF_LOG_ERR("error: temp collect failed");
+			reset();
+
+			/* collect fails, re-initiate a temperature read command
+			 * or we are stuck.
+			 */
+			if (_request(ADDR_CMD_CONVERT_D2) < 0) {
+				DF_LOG_ERR("error: temp measure failed");
+			}
 
-	// Request to convert the temperature
-	if (_request(ADDR_CMD_CONVERT_D2) < 0) {
-		DF_LOG_ERR("error: temp measure failed");
-		return;
-	}
+			return;
+		}
 
-	usleep(MS5611_CONVERSION_INTERVAL_US);
+		// Request to convert the pressure
+		if (_request(ADDR_CMD_CONVERT_D1) < 0) {
+			DF_LOG_ERR("error: pressure measure failed");
+		}
 
-	// read the result
-	uint32_t temperature_from_sensor;
+		_measure_phase++;
 
-	if (_collect(&temperature_from_sensor) < 0) {
-		DF_LOG_ERR("error: temp collect failed");
-		reset();
-		return;
-	}
+	} else {
+		if (_collect(pressure_from_sensor) < 0) {
+			DF_LOG_ERR("error: pressure collect failed");
+			reset();
 
-	// Request to convert the pressure
-	if (_request(ADDR_CMD_CONVERT_D1) < 0) {
-		DF_LOG_ERR("error: pressure measure failed");
-		return;
-	}
+			/* collect fails, re-initiate a pressure read command
+			 * or we are stuck.
+			 */
+			if (_request(ADDR_CMD_CONVERT_D1) < 0) {
+				DF_LOG_ERR("error: pressure measure failed");
```

Comment:
This return case should move out of  `if (_request(ADDR_CMD_CONVERT_D1) < 0) { }`.
When pressure collecting failed, do a reset and re-issue a request to collect again.
The logic is same with temperature collection above.